src/pudl/metadata/classes.py

```diff
@@ -1642,6 +1642,15 @@ def enforce_schema(self, df: pd.DataFrame) -> pd.DataFrame:
                 f"schema: {missing_cols}"
             )
 
+        # Log warning if columns in dataframe are getting dropped in write
+        dropped_columns = list(df.columns.difference(expected_cols))
+        if dropped_columns:
+            logger.warning(
+                "The following columns are getting dropped when writing to SQLite:"
+                f"{dropped_columns}. To keep these columns, add them to the "
+                f"metadata.resources fields and update alembic."
+            )
+
```

Comment:
My recollection is that there are many tables in which we intentionally rely on `enforce_schema()` to remove extra columns that aren't part of the table. In which case this change will create a large number of spurious warnings that should actually be ignored, making it hard to catch cases where it's actually a problem and maybe confusing people (later us).

Of course we don't want to be accidentally dropping columns we mean to keep, but is there a less noisy / more targeted way we can do that? The change to being stricter about the column mappings we made recently should help.

Maybe we could make it `logger.info()` instead, and make it clear in the message that we often intend this behavior.